loadFromBiom bugfix

TuomasBorman opened the pull request:
Hi,

I found a bug. Additionally, I polished the code little bit

TuomasBorman committed:
up

TuomasBorman commented on R/makeTreeSummarizedExperimentFromBiom.R:
```diff
@@ -151,16 +142,13 @@ makeTreeSEFromBiom <- function(
-                                                         clean.taxa.names))
     }
+    # Clean feature_data from possible character artifacts
+    feature_data <- .detect_taxa_artifacts_and_clean(feature_data, clean.taxa.names)
```
Better to do this in this step --> the input is always data.frame and we save some code lines

ChouaibB replied:
Ahh you are right you mean moving the detection thingy after this [line](https://github.com/microbiome/mia/blob/master/R/makeTreeSummarizedExperimentFromBiom.R#L153) ?

ChouaibB replied:
or perhaps [here](https://github.com/microbiome/mia/blob/master/R/makeTreeSummarizedExperimentFromBiom.R#L164) and after that cast it to `DataFrame` ?

TuomasBorman commented on R/makeTreeSummarizedExperimentFromBiom.R:
```diff
@@ -151,16 +142,13 @@ makeTreeSEFromBiom <- function(
         })
         # Create a data.frame from the list
         feature_data <- do.call(rbind, feature_data)
-        # Transposing feature_data and make it DFrame object
-        feature_data <- DataFrame(t(feature_data))
```
Is transposing necessary? This led to an error (my data was in correct order already)

ChouaibB replied:
That's what I was struggling with I don't know how the data looks like when it is loaded as a list [here](https://github.com/microbiome/mia/blob/master/R/makeTreeSummarizedExperimentFromBiom.R#L98C58-L98C58) ? do you have some examples please?

TuomasBorman commented on R/makeTreeSummarizedExperimentFromBiom.R:
```diff
@@ -280,30 +267,18 @@ makeTreeSummarizedExperimentFromBiom <- function(obj, ...){
-                 x_sub %>% stringr::str_split("") %>% unlist(),
-                 invert = invert, value = TRUE) %>% unique()
-        }) %>% unlist() %>% unique() %>% paste0(collapse = "")
-    }
```
Removed separate functionality since now there are only data.frames

TuomasBorman commented on R/makeTreeSummarizedExperimentFromBiom.R:
```diff
@@ -280,30 +267,18 @@ makeTreeSummarizedExperimentFromBiom <- function(obj, ...){
-        x <- lapply(x, gsub, pattern = patterns, replacement = "")
-    } else if (is(x, "data.frame")) {
-        x <- apply(x, 2, gsub, pattern = patterns, replacement = "")
-    }
```
Input can only be data.frame

ChouaibB replied:
True after I change the place where I cast all to data.frame i'll update the utility functions

TuomasBorman commented on tests/testthat/test-4IO.R:
```diff
@@ -11,6 +11,116 @@ test_that("Importing biom files yield SummarizedExperiment objects", {
     expect_s4_class(me2, "SummarizedExperiment")
     expect_equal(dim(me), dim(me2))
     expect_equal(rowData(me), rowData(me2))
+    
+    biom_object <- biomformat::read_biom(
+        system.file("extdata/testdata/Aggregated_humanization2.biom",
```
This file is correct place for tests --> moved them here

ChouaibB replied:
Sorry, I didn't understand the comment I didn't touch that file :)

ChouaibB commented:
I need more coffee :/ , I just realized that you did the whole fixing and work, I first thought you requested me to do the changes. 
Thanks Tuomas all looks good for me.

TuomasBorman commented:
> I need more coffee :/ , I just realized that you did the whole fixing and work, I first thought you requested me to do the changes. Thanks Tuomas all looks good for me.

:D yes I was fast --> good, I will still check this through

ChouaibB commented:
Thanks a lot :)

TuomasBorman committed:
up

TuomasBorman committed:
up

TuomasBorman commented:
I simplified the code, check if it is still good.

I removed warnings since there are also no warnings when prefixes are removed. Or do you think they are necessary?

TuomasBorman committed:
up

TuomasBorman committed:
up

TuomasBorman committed:
up

TuomasBorman committed:
Merge branch 'master' into loadfrombiom_bugfix

ChouaibB commented:
Removing the warnings is ok I guess as long as all is parsed well. I think at the end of the day the user itself should do some double checks after building/importing the object.
All good :) 
Thanks :)
